### PR TITLE
Game center auth

### PR DESCRIPTION
**Pull request by pragatimodi**
Adding signing in with game center auth functionality

### CODE REVIEW

**Comment by renkelvin** [comment]
Please fix the CI failure.

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExampleUITests/AuthenticationExampleUITests.swift [hunk 1]
```diff
@@ -36,10 +36,11 @@ class AuthenticationExampleUITests: XCTestCase {
     XCTAssertTrue(app.navigationBars["Firebase Auth"].exists)
   }
 
-  func testAuthOptions() {
-    // There are 15 sign in methods, each with its own cell
-    XCTAssertEqual(app.tables.cells.count, 15)
-  }
+  // TODO: Modify this test after code refactoring, current AuthMenu items aren't necessarily sign in methods
```
Comment (paulb777) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExampleUITests/AuthenticationExampleUITests.swift:39]: I'd be ok to delete.

**Review by paulb777 — APPROVED** [review]
Thanks! 

Any assessment of the Game Center functionality based on the new sample code?

**Comment by pragatimodi** [comment]
Tested by @renkelvin using an Apple developer account, it is working as expected.

**Pull request merged**